packages/oats-runtime/src/make.ts

```diff
@@ -428,10 +440,31 @@ export function makeObject<
       }
       result[index] = propResult.success();
     }
+
+    // tag each constructed object with a hidden type property
+    Object.defineProperty(result, reflection, { enumerable: false, value: type });
+
     return Make.ok(result);
   };
 }
 
+/**  Get reflection type from a  made value.
+ * Note that only directly made object values or ValueClasses can be used
+ * */
+export function getType(value: Record<string, any>): Type | undefined {
```

Comment:
this might be useful for other metacoding trickery also as we do not now need a named top level valueclass to know what is the type of any made object.